test/server.test.js

```diff
@@ -229,6 +229,33 @@ test('rm route and clear cached route', function (t) {
     });
 });
 
+test('rm one version of the routes, other versions should still work',
+    function (t) {
+        SERVER.get({ path: '/hello/:name', version: '1.0.0'},
+            function (req, res, next) {
+                res.send('hello ' + req.params.name);
+                next();
+            });
+        SERVER.get({ path: '/hello/:name', version: '2.0.0'},
+            function (req, res, next) {
+                res.send('hello ' + req.params.name);
+                next();
+            });
+
+        var routeThree = SERVER.get({ path: '/hello/:name', version: '3.0.0'},
+            function (req, res, next) {
+                res.send('hello ' + req.params.name);
+                next();
+            });
+
+        t.ok(SERVER.rm(routeThree));
+
+        CLIENT.get('/hello/whatever', function (err, _, res) {
```

Comment:
Can we do a strict check here, and make a request specifically for both 1.0.0 and 2.0.0 to ensure they're both available? Should be able to add a header version to do that.